src/overlays/actors/ovl_En_Sw/z_en_sw.c

```diff
@@ -35,95 +33,1003 @@ const ActorInit En_Sw_InitVars = {
+    u32 sp5C;
+    Vec3f sp50;
+    s32 sp4C = 1;
+
+    if (this->collider.base.ocFlags1 & 2) {
+        this->collider.base.acFlags &= ~2;
+        sp4C = 0;
+    } else if (((globalCtx->state.frames % 4) == 0) &&
+               (BgCheck_EntityLineTest1(&globalCtx->colCtx, &this->actor.world.pos, &this->unk_454, &sp50, &sp60, 1, 0,
+                                        0, 1, &sp5C) == 0)) {
+        sp4C = 0;
+    } else if (((globalCtx->state.frames % 4) == 1) &&
+               (BgCheck_EntityLineTest1(&globalCtx->colCtx, &this->actor.world.pos, &this->unk_460, &sp50, &sp60, 1, 0,
+                                        0, 1, &sp5C) != 0)) {
+        sp4C = 0;
+    } else if (((globalCtx->state.frames % 4) == 2) &&
+               (BgCheck_EntityLineTest1(&globalCtx->colCtx, &this->actor.world.pos, &this->unk_46C, &sp50, &sp60, 1, 0,
+                                        0, 1, &sp5C) == 0)) {
+        if (0) {}
+        sp4C = 0;
+    } else if (((globalCtx->state.frames % 4) == 3) &&
+               (BgCheck_EntityLineTest1(&globalCtx->colCtx, &this->actor.world.pos, &this->unk_478, &sp50, &sp60, 1, 0,
+                                        0, 1, &sp5C) != 0)) {
+        sp4C = 0;
+    }
+
+    if (BgCheck_EntityLineTest1(&globalCtx->colCtx, &this->actor.world.pos, &this->unk_484, &sp50, &this->unk_430, 1, 0,
+                                0, 1, &sp5C) != 0) {
+        this->actor.wallYaw = Math_FAtan2F(this->unk_430->normal.x, this->unk_430->normal.z) * 10430.378f;
+        this->actor.world.pos = sp50;
+        this->actor.world.pos.x += (6.0f * Math_SinS(this->actor.world.rot.y));
+        this->actor.world.pos.z += (6.0f * Math_CosS(this->actor.world.rot.y));
+        this->unk_434 = sp50;
+        this->unk_434.x += Math_SinS(this->actor.world.rot.y);
+        this->unk_434.z += Math_CosS(this->actor.world.rot.y);
+    }
+
+    return sp4C;
+}
+
+void func_80B0E314(EnSw* this, Vec3f arg1, f32 arg4) {
+    f32 xDist;
+    f32 yDist;
+    f32 zDist;
+    f32 dist;
+    f32 xDiff;
+    f32 yDiff;
+    f32 zDiff;
+
+    Math_SmoothStepToF(&this->actor.speedXZ, arg4, 0.3f, 100.0f, 0.1f);
+    xDiff = arg1.x - this->actor.world.pos.x;
+    yDiff = arg1.y - this->actor.world.pos.y;
+    zDiff = arg1.z - this->actor.world.pos.z;
+    dist = sqrtf(SQ(xDiff) + SQ(yDiff) + SQ(zDiff));
+    if (dist == 0.0f) {
+        xDist = yDist = zDist = 0.0f;
+    } else {
+        xDist = xDiff / dist;
+        yDist = yDiff / dist;
+        zDist = zDiff / dist;
+    }
+    xDist *= this->actor.speedXZ;
+    yDist *= this->actor.speedXZ;
+    zDist *= this->actor.speedXZ;
+    this->actor.world.pos.x += xDist;
+    this->actor.world.pos.y += yDist;
+    this->actor.world.pos.z += zDist;
+}
+
+s32 func_80B0E430(EnSw* this, f32 arg1, s16 arg2, s32 arg3, GlobalContext* globalCtx) {
+    Camera* activeCam;
+    f32 lastFrame;
+
+    lastFrame = Animation_GetLastFrame(&D_06000304.common);
+    if (DECR(this->unk_388) != 0) {
```

Comment:
```suggestion
    f32 lastFrame = Animation_GetLastFrame(&D_06000304);

    if (DECR(this->unk_388) != 0) {
```
`.common` no longer needed.